src/test/java/nextstep/subway/path/application/PathServiceTest.java

```diff
@@ -0,0 +1,135 @@
+package nextstep.subway.path.application;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.line.domain.LineRepository;
+import nextstep.subway.line.domain.Section;
+import nextstep.subway.path.dto.PathRequest;
+import nextstep.subway.path.dto.PathResponse;
+import nextstep.subway.station.application.StationService;
+import nextstep.subway.station.domain.Station;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class PathServiceTest {
+
+    private Station 교대역;
+    private Station 강남역;
+    private Station 남부터미널역;
+    private Station 양재역;
+    private Line 이호선;
+    private Line 삼호선;
+    private Line 신분당선;
+
+    private PathService pathService;
+
+    @Mock
+    private StationService stationService;
+
+    @Mock
+    private LineRepository lineRepository;
+
+    /**
+     * 교대역   --- *2호선*(5) ---   강남역
+     * |                            |
+     * *3호선(3)*                 *신분당선*(5)
+     * |                            |
+     * 남부터미널역 --- *3호선*(2) --- 양재역
+     */
+    @BeforeEach
+    public void setUp() {
+        pathService = new PathService(lineRepository, stationService);
+
+        교대역 = new Station("교대역");
+        강남역 = new Station("강남역");
+        남부터미널역 = new Station("남부터미널역");
+        양재역 = new Station("양재역");
+
+        신분당선 = new Line("신분당선", "red", 강남역, 양재역, 5);
+        이호선 = new Line("이호선", "green", 교대역, 강남역, 5);
+        삼호선 = new Line("삼호선", "orange", 교대역, 양재역, 5);
+        삼호선.addSection(new Section(삼호선, 남부터미널역, 양재역, 2));
+    }
+
+    @Test
+    @DisplayName("교대역 -> 양재역의 최단 경로는 교대역 -> 남부터미널역 -> 양재역")
+    void findPathTest() {
+        // when
+        when(stationService.findStationById(1L)).thenReturn(교대역);
+        when(stationService.findStationById(4L)).thenReturn(양재역);
+        when(lineRepository.findAll()).thenReturn(Arrays.asList(이호선, 삼호선, 신분당선));
+
+        PathResponse pathResponse = pathService.findPath(new PathRequest(1L, 4L));
+
+        // then
+        assertAll(
+                () -> assertThat(pathResponse.getStations()).hasSize(3),
+                () -> assertThat(pathResponse.getDistance()).isEqualTo(5)
+        );
+    }
+
+    @Test
+    @DisplayName("조회하려는 최단 구간의 출발역과 도착역은 달라야 한다")
+    void 출발역과_도착역이_같은_경우() {
+        // when
+        when(stationService.findStationById(1L)).thenReturn(교대역);
+        when(lineRepository.findAll()).thenReturn(Arrays.asList(이호선, 삼호선, 신분당선));
```

Comment:
84번째 줄의 주석이 // when 으로 기록이 되어 있는데요,
보통은 given - when - then라고 합니다.

given에선 테스트의 조건이나 환경을 설정하고 when에서 수행한 결과를 then에서 검증하는 것입니다.

**엥? 근데 왜 메소드는 when이죠?** 라는 의문을 품으실 수 있어요.
그래서 `org.mockito.BDDMockito.given`을 많이들 쓰고 있습니다.
덜 헷갈려서 좋거든요.

```suggestion
        // given
        given(stationService.findStationById(1L)).willReturn(교대역);
        given(lineRepository.findAll()).willReturn(Arrays.asList(이호선, 삼호선, 신분당선));
```

Reply:
각 객체의 결과가 해당 값일때 라는 의미로 when에 넣었었는데 그 보단 주어지는 값이라는 의미로 given에 넣는것이 맞는거군요 😲`org.mockito.BDDMockito.given` 을 적용해서 수정해보겠습니다!